enhancements/machine-config/machine-config-node.md

```diff
@@ -0,0 +1,525 @@
+    Message:               This node has not yet entered the AppliedFilesAndOS phase
+    Reason:                NotYetOccured
+    Status:                False
+    Type:                  AppliedFilesAndOS
+    Last Transition Time:  2023-11-22T20:08:48Z
+    Message:               Cordoned node. The node is reporting Unschedulable = true
+    Reason:                UpdateExecutedCordoned
+    Status:                True
+    Type:                  Cordoned
+    Last Transition Time:  2023-11-22T20:05:26Z
+    Message:               This node has not yet entered the RebootedNode phase
+    Reason:                NotYetOccured
+    Status:                False
+    Type:                  RebootedNode
+    Last Transition Time:  2023-11-22T20:05:26Z
+    Message:               This node has not yet entered the ReloadedCRIO phase
+    Reason:                NotYetOccured
+    Status:                False
+    Type:                  ReloadedCRIO
+    Last Transition Time:  2023-11-22T20:08:46Z
+    Message:               Node ip-10-0-52-193.ec2.internal needs an update
+    Reason:                Updated
+    Status:                False
+    Type:                  Updated
+  Config Version:
+    Current:            rendered-worker-7f183a799b1bce71eca1b49fa20c7261
+    Desired:            rendered-worker-3b7e2ff302f33d45a732e03564405ae7
+  Observed Generation:  2
+Events:                 <none>
+```
+
+There are two levels of conditions in the MachineConfigNode type: the parent and the child condition. Parent conditions incude Updated, UpdatePrepared, UpdateExecuted, UpdatePostAction, UpdateComplete, and Resumed. These parent conditions track the overall arc of an upgrade. However, there are often multiple phases within these overarching ones. Therefore: UpdateCompatible, DrainedNode, ApplyingFilesAndOS, CordonedNode, RebootedNode, ReloadedCRIO, and UncordonedNode are the ChildrenPhases that occur during the larger ones. These don't always occur depending on the type of update. However, sometimes they all occur, leading to some confusion in the parent phases as to what happened. Adding thse children phases is meant to give the user more clsrity to the update at hand.
+
+For example. When a CRIO reload is not required on a node, the condition will look like this: 
+
+```console
+    Last Transition Time:  2024-01-12T14:47:21Z
+    Message:               This node has not yet entered the ReloadedCRIO phase
+    Reason:                NotYetOccured
+    Status:                False
+    Type:                  ReloadedCRIO
+```
+
+or 
+
+```console
+    Last Transition Time:  2024-01-12T14:52:19Z
+    Message:               Action during update to rendered-master-eef3a1a1a422b7ebd085b346042ca5a8: Upgrade required a CRIO reload. Completed this as the post update action.
+    Reason:                UpdatePostActionCompleteReloadedCRIO
+    Status:                False
+    Type:                  ReloadedCRIO
```

Comment:
IMO, these three are the important ones to align, the message is the possible outlier.

I think something like below I'd be ok with, since the message is clearly showing history, but the reason, status and type all align. Might want some more opinions on this though.
```
    Last Transition Time:  2024-01-12T14:52:19Z
    Message:               Last action: Update to rendered-master-eef3a1a1a422b7ebd085b346042ca5a8: Upgrade required a CRIO reload. Reload completed as a post update action.
    Reason:                UpdateIdle
    Status:                False
    Type:                  ReloadedCRIO
```

Reply:
hmmm ok so the reason is the main thing throwing this off? This is how it will most likely be in 4.15/16 but I can make a card in the follow up epic to change this?

Reply:
Yeah the reason is jarring to me as it's written now, it doesn't make a lot of sense. Has my operator reloaded crio? No, status is false, ok, why not, UpdatePostActionCompleteReloadedCrio, well that sounds like it has reloaded crio, so status should be true